### PR TITLE
Correct exitcode when running through `subprocess.run`

### PR DESCRIPTION
**Pull request by moreal**
This pull request fixes exitcode of `SIGINT` (i.e., `KeyboardInterrupt`) when running through `subprocess.run`.

You can run the below script.

```python
import sys, subprocess; subprocess.run([sys.executable, "-c", "raise KeyboardInterrupt"])
```

## CPython

```python
Python 3.10.9 (main, Dec 15 2022, 17:11:09) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys, subprocess; subprocess.run([sys.executable, "-c", "raise KeyboardInterrupt"])
Traceback (most recent call last):
  File "<string>", line 1, in <module>
KeyboardInterrupt
CompletedProcess(args=['/opt/homebrew/opt/python@3.10/bin/python3.10', '-c', 'raise KeyboardInterrupt'], returncode=-2)
```

## Before (RustPython main)

```python
>>>>> import sys, subprocess; subprocess.run([sys.executable, "-c", "raise KeyboardInterrupt"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyboardInterrupt
CompletedProcess(args=['/path/to/RustPython/RustPython/target/debug/rustpython', '-c', 'raise KeyboardInterrupt'], returncode=130)
```

## After (RustPython)

```python
Welcome to the magnificent Rust Python 0.2.0 interpreter 😱 🖖
>>>>> import sys, subprocess; subprocess.run([sys.executable, "-c", "raise KeyboardInterrupt"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyboardInterrupt
CompletedProcess(args=['/path/to/RustPython/RustPython/target/debug/rustpython', '-c', 'raise KeyboardInterrupt'], returncode=-2)
```

## Related links

 - #4414 
 - https://github.com/python/cpython/blob/8b1f1251215651c4ef988622345c5cb134e54d69/Modules/main.c#L652-L657

### CODE REVIEW

**Comment by moreal** [comment]
I researched about failing in `test_subprocess` and it was because it kills the process before flushing `sys.stderr` and `sys.stdout`. I fixed it by inlining and flushing them before killing process.

File: Lib/test/test_runpy.py [hunk 1]
```diff
@@ -843,12 +833,23 @@ def test_pymain_run_command(self):
...
+            test_pymain_run_file,
+            test_pymain_run_file_runpy_run_module,
+            test_pymain_run_file_runpy_run_module_as_main,
+            test_pymain_run_command_run_module,
+            test_pymain_run_command,
+            test_pymain_run_module
+        ]
+        for t in windows_only_failed_tests:
+            unittest.expectedFailure(t)
+
```
Comment (fanninpm) [inline Lib/test/test_runpy.py:852]: 👍

File: vm/src/vm/mod.rs [hunk 2]
```diff
@@ -763,6 +768,22 @@ impl VirtualMachine {
...
+                                SigSet::empty(),
+                            ),
+                        )
+                        .is_ok()
+                        {
+                            interpreter::flush_std(self);
+                            kill(getpid(), SIGINT).expect("Expect to be killed.");
+                        }
+                    }
+
```
Comment (fanninpm) [inline vm/src/vm/mod.rs:786]: This seems like a lot of `unsafe`.

---

Does Rust allow, e.g.
```rust
if (unsafe { dangerous() }) {
    // ...
}
```

…or is the syntax too restrictive for that?

**Review by fanninpm — APPROVED** [review]
LGTM

File: vm/src/vm/mod.rs (hunk 2 above)
Comment (moreal) [reply vm/src/vm/mod.rs:786]: I didn't know the syntax is available. Only `sigaction` function needs `unsafe`. So I'll apply it like with `cargo fmt`.

```suggestion
                        if unsafe { sigaction(
                            SIGINT,
                            &SigAction::new(
                                nix::sys::signal::SigHandler::SigDfl,
                                SaFlags::SA_ONSTACK,
                                SigSet::empty(),
                            ),
                        ) }
                        .is_ok()
                        {
                            interpreter::flush_std(self);
                            kill(getpid(), SIGINT).expect("Expect to be killed.");
                        }

```

Comment (youknowone) [reply vm/src/vm/mod.rs]: in general, you may want to make a new variable for complex expressions.
```
let r = unsafe { sigaction(
                            SIGINT,
                            &SigAction::new(
                                nix::sys::signal::SigHandler::SigDfl,
                                SaFlags::SA_ONSTACK,
                                SigSet::empty(),
                            ),
                        ) };
if r.is_ok() {
    ...
}
```

Comment (moreal) [reply vm/src/vm/mod.rs]: I applied it in [a3507e7](https://github.com/RustPython/RustPython/pull/4454/commits/a3507e7f58aaa13897cb5d15608ab4499f50f415) commit 🙏🏻

**Comment by youknowone** [comment]
Thank you!

**Pull request merged**